pkg/cosign/tuf/client.go

```diff
@@ -247,6 +261,47 @@ func (t *TUF) GetTarget(name string) ([]byte, error) {
 	return targetBytes, nil
 }
 
+// Get target files by a custom usage metadata tag. If there are no files found,
+// use the fallback target names to fetch the targets by name.
+func (t *TUF) GetTargetsByMeta(usage UsageKind, fallbacks []string) ([]TargetFile, error) {
+	targets, err := t.client.Targets()
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting targets")
+	}
+	var matchedTargets []TargetFile
+	for name, targetMeta := range targets {
+		// Skip any targets that do not include custom metadata.
+		if targetMeta.Custom == nil {
+			continue
+		}
+		var scm sigstoreCustomMetadata
+		err := json.Unmarshal(*targetMeta.Custom, &scm)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "**Warning** Custom metadata not configured properly for target %s, skipping target\n", name)
+			continue
+		}
+		if scm.Sigstore.Usage == usage {
+			target, err := t.GetTarget(name)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "**Warning** Unable to fetch target %s, skipping target\n", name)
```

Comment:
This feels like a legit error: `GetTargets` gives the string but it fails download? 
Or are you anticipating that maybe there's a difference between delegated targets and better safe than sorry?

Reply:
Yea, I went back and forth on this. I think it's fair to say that by this point, an error would be caused by something out of the user's control and therefore we should fail.
FWIW, we'll remove fallbacks after root v3, so we can make both of these cases errors then.